### PR TITLE
Option for default event in DefaultAuthenticationEventPublisher

### PR DESCRIPTION
**Pull request by zeeshanadnan**
…Publisher

Fixes spring-projectsgh-7825

### CODE REVIEW

File: core/src/test/java/org/springframework/security/authentication/DefaultAuthenticationEventPublisherTests.java [hunk 1]
```diff
@@ -137,6 +137,28 @@ public void unknownFailureExceptionIsIgnored() {
...
+	@Test(expected = IllegalArgumentException.class)
+	public void defaultAuthenticationFailureEventClassSetNullThen() {
+		publisher = new DefaultAuthenticationEventPublisher();
+		publisher.setDefaultAuthenticationFailureEvent(null);
+	}
+
+	@Test
+	public void defaultAuthenticationFailureEventIsPublished() {
+		publisher = new DefaultAuthenticationEventPublisher();
+		Properties p = new Properties();
```
Comment (jzheaux) [inline core/src/test/java/org/springframework/security/authentication/DefaultAuthenticationEventPublisherTests.java]: Are these lines necessary that set up the additional exception mappings? I'm thinking that lines 149 thru 152 can be removed.

File: core/src/test/java/org/springframework/security/authentication/DefaultAuthenticationEventPublisherTests.java [hunk 2]
```diff
@@ -137,6 +137,28 @@ public void unknownFailureExceptionIsIgnored() {
...
+		publisher.setDefaultAuthenticationFailureEvent(AuthenticationFailureBadCredentialsEvent.class);
+		ApplicationEventPublisher appPublisher = mock(ApplicationEventPublisher.class);
+
+		publisher.setApplicationEventPublisher(appPublisher);
+		publisher.publishAuthenticationFailure(new AuthenticationException("") {
+		}, mock(Authentication.class));
+		verify(appPublisher).publishEvent(isA(AuthenticationFailureBadCredentialsEvent.class));
+	}
+
 	private static final class MockAuthenticationException extends
```
Comment (jzheaux) [inline core/src/test/java/org/springframework/security/authentication/DefaultAuthenticationEventPublisherTests.java]: Would you please add a test for a class that is missing the appropriate constructor? That will make sure to test that the exception is thrown appropriately.

**Comment by zeeshanadnan** [comment]
Hi @jzheaux added your change requests into a new commit.

**Comment by jzheaux** [comment]
Thanks @zeeshanadnan for the changes. In preparation for merging, will you please squash your commits and change the commit message to match [the contribution guidelines](https://github.com/spring-projects/spring-security/blob/master/CONTRIBUTING.md#format-commit-messages)?

**Comment by zeeshanadnan** [comment]
> Thanks @zeeshanadnan for the changes. In preparation for merging, will you please squash your commits and change the commit message to match [the contribution guidelines](https://github.com/spring-projects/spring-security/blob/master/CONTRIBUTING.md#format-commit-messages)?

@jzheaux Done

**Comment by jzheaux** [comment]
Thanks, @zeeshanadnan, for the PR! This is now merged into master via `51b9b2f693ec48127f2cc8554aa3edbc58fccc92`.

Note that I polished the commit message just a bit, though it will still show you as the contributor. I added a small polish via `653400edfabf52760bdda3c30c8b51283b256b15`.

**Pull request closed without merging**